### DISCUSSION TITLE
Replacing styelint, eslint and prettier with biomejs

### DISCUSSION THREAD

**Discussion by theskumar** [OpenTechFund/opentech.fund#4283] [General]
## Pros
- faster
- Rust-based, Node.js not required
- Plays nicely with pre-commit, the current setup feel hacky
- Similar configuration, don’t have to work with three different tools 
- It’s built as a replacement for Prettier, ESLint, and the rules can be ported over
- Good editor support across VSCode

## Cons
- It’s a bit new
- While it supports almost all the rules, it doesn't support 100% of all the rules 

## Conclusion: 
Given this project is not FE heavy, we probably don't need a heavily customized tool to run linter and formatter; the current tools are quite slow, esp. This feels to me like the `ruff` of the Node.js world, and will help with the dev efficiency because we only have to work with a single tool that is easy to maintain, is faster, and provides better error reports. 

https://biomejs.dev/

cc: @frjo @wes-otf

**Comment by frjo** [comment]
I think it is a good idea to test this, using three tools is a bit messy.

I would prefer if we configure biome so there are minimal code changes done, at least in the first PR. Running it with default settings makes biome wanting to change every file it seems.

Configuring it so minimal changes are done to current code makes it easy to evaluate the tool itself and revert back easily if needed.

Then overtime we can adjust the settings to introduce code style improvements. (I think many of its suggestions are improvements.)
